src/reanimated2/interpolation.js

```diff
@@ -1,30 +1,115 @@
 import { Extrapolate } from '../derived/interpolate';
 
+function getVal(config) {
+  'worklet';
+
+  const { type, coef, val, ll, rr, x } = config;
+
+  switch (type) {
+    case Extrapolate.IDENTITY:
+      return x;
+    case Extrapolate.CLAMP:
+      if (coef * val < coef * ll) {
+        return ll;
+      }
+      return rr;
+    case Extrapolate.EXTEND:
+    default:
+      return val;
+  }
+}
+
+function isExtrapolate(value) {
+  'worklet';
+
+  return (
+    value === Extrapolate.EXTEND ||
+    value === Extrapolate.CLAMP ||
+    value === Extrapolate.IDENTITY
+  );
+}
+
+function validateType(type) {
+  'worklet';
+
+  if (!type && type !== undefined) {
+    throw new Error(
+      `Reanimated: config object is not valid please provide valid config, for example:
+       interpolate(value, [inputRange], [outputRange], {
+        extrapolateLeft: 'clamp',
+        extrapolateRight: 'extend',
+      }) or interpolate(value, [inputRange], [outputRange], 'clamp)`
+    );
+  }
+
+  if (
+    type?.constructor === Object &&
+    ((Object.keys(type).length === 2 &&
+      !('extrapolateLeft' in type && 'extrapolateRight' in type)) ||
+      (Object.keys(type).length === 1 &&
+        !('extrapolateLeft' in type || 'extrapolateRight' in type)) ||
+      Object.keys(type).length > 2)
```

Comment:
Maybe it could be done using less code?
How about:
```
type?.constructor === Object &&
    ((Object.keys(type).length >= 1 &&
      !('extrapolateLeft' in type && 'extrapolateRight' in type)) ||
    Object.keys(type).length > 2)
```
or
```
type?.constructor === Object &&
    (([1,2].includes(Object.keys(type).length) &&
      !('extrapolateLeft' in type && 'extrapolateRight' in type)) ||
    Object.keys(type).length > 2)
```

Reply:
Yep, looks too much anyway I ended up with this

Your code will not work for this code

In case if one of `extrapolateLeft | extrapolateRight` field is missing (by default if one of them is missing it's `extrapolate === Extrapolate.EXTEND`)

```js
interpolate(translation.y.value, [0, -75], [0, -150], {
        extrapolateLeft: 'clamp',
})
```

Reply:
Yes, you are right, sorry, I missed one thing in those conditions, it's ok.